cmd/snap-update-ns/change.go

```diff
@@ -339,8 +339,8 @@ func NeededChanges(currentProfile, desiredProfile *osutil.MountProfile) []*Chang
 	}
 
 	// Sort both lists by directory name with implicit trailing slash.
-	sort.Sort(byMagicDir(current))
-	sort.Sort(byMagicDir(desired))
+	sort.Sort(byOriginAndMagicDir(current))
+	sort.Sort(byOriginAndMagicDir(desired))
```

Comment:
So one comment on this. I think it's not incorrect in principle. I have some deeper thoughts but the point I want to convey is that the way overname _must_ work is incompatible with simply sorting by name. I'm continuing the investigation on how mount operation sorting ought to work to always be correct or to detect conflicts